lib/matplotlib/scale.py

```diff
@@ -128,79 +115,118 @@ def __str__(self):
         return "{}()".format(type(self).__name__)
 
 
+@cbook.deprecated("3.1", alternative="LogTransform")
 class Log10Transform(LogTransformBase):
     base = 10.0
 
     def inverted(self):
         return InvertedLog10Transform()
 
 
+@cbook.deprecated("3.1", alternative="InvertedLogTransform")
 class InvertedLog10Transform(InvertedLogTransformBase):
     base = 10.0
 
     def inverted(self):
         return Log10Transform()
 
 
+@cbook.deprecated("3.1", alternative="LogTransform")
 class Log2Transform(LogTransformBase):
     base = 2.0
 
     def inverted(self):
         return InvertedLog2Transform()
 
 
+@cbook.deprecated("3.1", alternative="InvertedLogTransform")
 class InvertedLog2Transform(InvertedLogTransformBase):
     base = 2.0
 
     def inverted(self):
         return Log2Transform()
 
 
+@cbook.deprecated("3.1", alternative="LogTransform")
 class NaturalLogTransform(LogTransformBase):
     base = np.e
 
     def inverted(self):
         return InvertedNaturalLogTransform()
 
 
+@cbook.deprecated("3.1", alternative="InvertedLogTransform")
 class InvertedNaturalLogTransform(InvertedLogTransformBase):
     base = np.e
 
     def inverted(self):
         return NaturalLogTransform()
 
 
-class LogTransform(LogTransformBase):
+class LogTransform(Transform):
+    input_dims = 1
+    output_dims = 1
+    is_separable = True
+    has_inverse = True
+
     def __init__(self, base, nonpos='clip'):
-        LogTransformBase.__init__(self, nonpos)
+        Transform.__init__(self)
         self.base = base
+        self._clip = {"clip": True, "mask": False}[nonpos]
+
+    def __str__(self):
+        return "{}(base={}, nonpos={!r})".format(
+            type(self).__name__, self.base, "clip" if self._clip else "mask")
+
+    def transform_non_affine(self, a):
+        # Ignore invalid values due to nans being passed to the transform.
+        with np.errstate(divide="ignore", invalid="ignore"):
+            log = {np.e: np.log, 2: np.log2, 10: np.log10}.get(self.base)
+            if log:  # If possible, do everything in a single call to Numpy.
+                out = log(a)
```

Comment:
Is this the optimization you are referring to?  I guess I think this is a good idea since it covers 99.9% of the cases, and would instead suggest changing the images or increasing the tolerance.  But I'm not an expert on how expensive the optimization is...